Restore 100,000 limit in HTMLOptionsCollection.length setter

Ahmad-S792 opened the pull request:
#### a27ac64f683e8d0a798123758cb4bf5a0bc75cd0
<pre>
Restore 100,000 limit in HTMLOptionsCollection.length setter

<a href="https://bugs.webkit.org/show_bug.cgi?id=252983">https://bugs.webkit.org/show_bug.cgi?id=252983</a>

Reviewed by Chris Dumez.

This patch is to align WebKit with Blink / Chromium and Gecko / Firefox.

Merge - <a href="https://chromium.googlesource.com/chromium/src/+/f27e6ea87ecf211c8b8644813422a9e7f19cd1cc">https://chromium.googlesource.com/chromium/src/+/f27e6ea87ecf211c8b8644813422a9e7f19cd1cc</a>

This patch updates &apos;maxSelectItems&apos; to new value of 100,000
to reflect update in spec.
Further, it is updated to be only used when new length is
greater than current length.
Additionally, add comments to reflect the details as needed.

Web-Spec: <a href="https://html.spec.whatwg.org/#dom-htmloptionscollection-length">https://html.spec.whatwg.org/#dom-htmloptionscollection-length</a>
Issue: <a href="https://github.com/whatwg/html/issues/8337">https://github.com/whatwg/html/issues/8337</a>

* Source/WebCore/html/HTMLSelectElement.cpp:
(maxSelectItems): Update constant value
(HTMLSelectElement::setItem): Remove &apos;=&apos; and update comments and console message
(HTMLSelectElement::setLength): Add comment and update console message
* LayoutTests/imported/w3c/web-platform-tests/html/select/options-length-too-large.html: Add Test Case
* LayoutTests/imported/w3c/web-platform-tests/html/select/options-length-too-large-expected.txt: Add Test Case Expectation
* LayoutTests/fast/forms/select-max-length-expected.txt: Rebaselined
* LayoutTests/fast/dom/HTMLSelectElement/select-selectedIndex-multiple-expected.txt: Rebaselined
* LayoutTests/fast/dom/HTMLSelectElement/select-selectedIndex-expected.txt: Rebaselined

Canonical link: <a href="https://commits.webkit.org/260896@main">https://commits.webkit.org/260896@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/feee81464c78d20f33687f74b884a1f5b9d6d93f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109855 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18965 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/42573 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1312 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/118945 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/113805 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20431 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10160 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102123 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115603 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/15211 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/42573 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/43433 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/97174 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/42573 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/85235 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11677 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/42573 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/12320 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/8374 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/17689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/42573 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7553 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14089 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/html/HTMLSelectElement.cpp:
```diff
@@ -66,8 +66,8 @@ using namespace WTF::Unicode;
-// Upper limit agreed upon with representatives of Opera and Mozilla.
-static const unsigned maxSelectItems = 10000;
+// https://html.spec.whatwg.org/#dom-htmloptionscollection-length
+static const unsigned maxSelectItems = 100000;
```
`constexpr` is preferred.

Ahmad-S792 replied:
Will update after EWS run to confirm all tests are now passing so I can do rebase and other bits in one go. Thanks for review!

Ahmad-S792 replied:
Done.

Ahmad-S792 committed:
Restore 100,000 limit in HTMLOptionsCollection.length setter

https://bugs.webkit.org/show_bug.cgi?id=252983

Reviewed by Chris Dumez.

This patch is to align WebKit with Blink / Chromium and Gecko / Firefox.

Merge - https://chromium.googlesource.com/chromium/src/+/f27e6ea87ecf211c8b8644813422a9e7f19cd1cc

This patch updates 'maxSelectItems' to new value of 100,000
to reflect update in spec.
Further, it is updated to be only used when new length is
greater than current length.
Additionally, add comments to reflect the details as needed.

Web-Spec: https://html.spec.whatwg.org/#dom-htmloptionscollection-length
Issue: https://github.com/whatwg/html/issues/8337

* Source/WebCore/html/HTMLSelectElement.cpp:
(maxSelectItems): Update constant value
(HTMLSelectElement::setItem): Remove '=' and update comments and console message
(HTMLSelectElement::setLength): Add comment and update console message
* LayoutTests/imported/w3c/web-platform-tests/html/select/options-length-too-large.html: Add Test Case
* LayoutTests/imported/w3c/web-platform-tests/html/select/options-length-too-large-expected.txt: Add Test Case Expectation
* LayoutTests/fast/forms/select-max-length-expected.txt: Rebaselined
* LayoutTests/fast/dom/HTMLSelectElement/select-selectedIndex-multiple-expected.txt: Rebaselined
* LayoutTests/fast/dom/HTMLSelectElement/select-selectedIndex-expected.txt: Rebaselined

Canonical link: https://commits.webkit.org/260896@main
